File: samples/simple-drive-app/simple-drive-app/Controllers/FilesTableViewController.m
/*******************************************************************************
 * Copyright (c) Microsoft Open Technologies, Inc.
 * All Rights Reserved
 * See License.txt in the project root for license information.
 ******************************************************************************/

#import "BaseController.h"
#import "FilesTableViewController.h"


@interface FilesTableViewController()

@property NSArray *Files;
@property MSSharePointClient* client;

@end

@implementation FilesTableViewController

- (id)initWithStyle:(UITableViewStyle)style
{
    self = [super initWithStyle:style];
    
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [BaseController getClient:^(MSSharePointClient * client) {
        
        self.client = client;
        [self getFiles];
    }];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.Files.count;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"FileCell" forIndexPath:indexPath];
    
    MSSharePointItem *file = (MSSharePointItem*)[self.Files objectAtIndex:indexPath.row];
    
    cell.textLabel.text = file.name;
    
    return cell;
}

-(void)getFiles {
    
    UIActivityIndicatorView *spinner = [BaseController getSpinner:self.view];
    
    
    [self.client.files readWithCallback:^(NSArray *items, MSOrcError *error) {
        
        if (error == nil) {
            
            dispatch_async(dispatch_get_main_queue(),
                           ^{
                               self.Files = items;
                               [self.tableView reloadData];
                               [spinner stopAnimating];
                           });
        }
        else{
            dispatch_async(dispatch_get_main_queue(),
                           ^{
                               [spinner stopAnimating];
                               
                               UIAlertView* alert = [[UIAlertView alloc] initWithTitle:@"Error!" message:[[error userInfo] objectForKey:0] delegate:self cancelButtonTitle:@"Ok" otherButtonTitles:nil, nil];
                               [alert show];
                           });
        }
    }];
}

- (IBAction)unwindExchangeViews:(UIStoryboardSegue *)segue{
}

- (IBAction)Refresh:(id)sender {
    [self getFiles];
}

- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath
{
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        
        __block UIActivityIndicatorView *spinner = [BaseController getSpinner:self.view];
        
        MSSharePointFile *fileToDelete = [self.Files objectAtIndex:indexPath.row];
        
        [[self.client.files getById:fileToDelete.id] delete:^(int status, MSOrcError *error) {
            
            dispatch_async(dispatch_get_main_queue(),
                           ^{
                               
                               [spinner stopAnimating];
                               [self getFiles];
                               
                               UIAlertView* alert = [[UIAlertView alloc] initWithTitle:@"Success"
                                                                               message:[@"Deleted File " stringByAppendingString:fileToDelete.name]
                                                                              delegate:self cancelButtonTitle:@"Ok" otherButtonTitles:nil, nil];
                               [alert show];
                           });
            
        }];
        
    }
}

@end

File: sdk-swift/samples/simple-files-app/simple-files-app/FilesTableViewController.swift
/*******************************************************************************
* Copyright (c) Microsoft Open Technologies, Inc.
* All Rights Reserved
* Licensed under the Apache License, Version 2.0.
* See License.txt in the project root for license information.
******************************************************************************/

import UIKit
import office365_files_sdk

class FilesTableViewController: UITableViewController {

    var files = NSMutableArray();
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.getFiles();
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {

        return self.files.count;
    }

    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("FileCell", forIndexPath: indexPath) as UITableViewCell
        var file = self.files.objectAtIndex(indexPath.row) as MSSharePointItem;
        var id : NSString = file.id;
        cell.textLabel.text = file.name;
        return cell
    }

    func getFiles(){
    
        var spinner = BaseController.getSpinner(self.view);

        BaseController.getClient({ (client) -> Void! in
            client.getFiles().read({ (entities : Array<MSSharePointItem>?, error) -> Void! in
        
                dispatch_async(dispatch_get_main_queue(), { () -> Void in
                    
                    spinner.stopAnimating();
                    self.files.addObjectsFromArray(entities!);
                    self.tableView.reloadData();
                });
                
                return Void();
                
            }).resume();
        });
    }
}